charmcraft.yaml

```diff
@@ -163,3 +166,7 @@ actions:
         description: >-
           Backup ID to restore.
     required: [backup-id]
+  
+  get-ingress-url:
+    description: >-
+      Get the ingress URL for the charm.
```

Comment:
Why add this action? And why here and not on the k8s charm. It feels like this action belongs in the ingress charm, and not here. For instance, traefik has a `show-proxied-endpoints` action

Reply:
That's a valid concern, the answer could be yes. I'd say there isn't much logic here that needs to belong in one specific place, it's more about the UX.
In ideal world I'd say yes, all charms implementing the requirer side of the ingress integration should have it.
If I was to operate one specific charm for example Vault in a large enterprise with a lot of charms deployed I'd prefer to run the action against the charm I am operating and get the answer I need instead of running it against the charm operated by another team and get a list of all proxied endpoint then filter them to find the one I am looking for.

But at the end this can be a subjective thing and a question of preference (What's the best UX in this case?)

Reply:
Action removed. This affects the integration tests, for now the test case is really minimal, we can extend it once there is an action in HAProxy to get the ingress URL.